# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalResponse.h
# Commit: 66b279e5a513
# Full Hash: 66b279e5a513bf706129ac64cf44e3353e3a0d71
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-12 21:30:33
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - FetchService integration for PFetch. r=dom-worker-reviewers,jesup
#   
#   Depends on D142436
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142437
# ==============================================================================

diff -r 3eb8fdd0ba6d -r 66b279e5a513 dom/fetch/InternalResponse.h
--- a/dom/fetch/InternalResponse.h	Thu Jan 12 17:24:28 2023 +0000
+++ b/dom/fetch/InternalResponse.h	Thu Jan 12 17:24:28 2023 +0000
@@ -311,6 +311,9 @@
 
   bool HasBeenCloned() const { return mCloned; }
 
+  void SetSerializeAsLazy(bool aAllow) { mSerializeAsLazy = aAllow; }
+  bool CanSerializeAsLazy() const { return mSerializeAsLazy; }
+
   void InitChannelInfo(nsIChannel* aChannel) {
     mChannelInfo.InitFromChannel(aChannel);
   }
@@ -380,6 +383,7 @@
   nsCOMPtr<nsIInputStream> mAlternativeBody;
   nsMainThreadPtrHandle<nsICacheInfoChannel> mCacheInfoChannel;
   bool mCloned;
+  bool mSerializeAsLazy{true};
 
  public:
   static constexpr int64_t UNKNOWN_BODY_SIZE = -1;